packages/plugin-pack/sources/packUtils.ts

```diff
@@ -67,9 +69,10 @@ export async function prepareForPack(workspace: Workspace, {report}: {report: Re
   }
 }
 
-export async function genPackStream(workspace: Workspace, files?: Array<PortablePath>) {
-  if (typeof files === `undefined`)
-    files = await genPackList(workspace);
+export async function genPackStream(workspace: Workspace, files?: Array<PortablePath | {file: PortablePath, source: PortablePath}>) {
+  const normalizedFiles = files?.map(file => typeof file === `string`
+    ? {file, source: ppath.resolve(workspace.cwd, file)}
```

Comment:
I'd suggest to make `source` a `string` parameter (and to read the content of the license file outside of the `genPackStream` function); this way it gives us a way to generate files within the archive that don't necessarily exist on disk (for example if we wanted to generate the LICENSE file from the SPDX code).

Reply:
I believe source is already is a `string` because `PortablePath` is in fact a `string`. I'm not too aware of the differences between the 2 though.

Also because I'm not too familiar with how gzip and tar works. Can give me further guidance on where the inclusion  of LICENSE files should happen instead? Thank you.